### ISSUE TITLE
KBFS basically nonexistent now, worked before

### ISSUE THREAD

**Issue by tcby45** [keybase/client#2679]
Hello there!

I installed KBFS on an instance of Ubuntu 14.04 a couple of days ago. I was able to successfully look at others' `/keybase/public` directories and even write my own `index.md` last night, but now, it doesn't work. Namely, there is no directory structure in `/keybase` at all.

```sh
ubuntu in ~(ruby-2.3.0) λ cd /keybase
ubuntu in /keybase(ruby-2.3.0) λ sudo ls -al
total 8
drwxrwxrwx  2 root root 4096 Apr 24 00:30 .
drwxr-xr-x 23 root root 4096 Apr 24 01:10 ..
ubuntu in /keybase(ruby-2.3.0) λ cd /keybase/public/readyp1
cd: no such file or directory: /keybase/public/readyp1
ubuntu in /keybase(ruby-2.3.0) λ cd /keybase/public/chris
cd: no such file or directory: /keybase/public/chris
```
I don't have access to a Windows/Mac machine at the moment, so I can't try dropping anything in there. What should I do?

my log id: 5e253f5060682886b6032d1c

**Comment by tcby45** [comment]
Just realized something - if it means anything, just after it **was** working, I did a `sudo reboot` before going to sleep, which may have messed things up.

**Comment by cjb** [comment]
Does `run_keybase` help?

I also see this: /opt/keybase/Keybase: error while loading shared libraries: libnotify.so.4: cannot open shared object file: No such file or directory

**Comment by tcby45** [comment]
Yes, it works again! Is there a way for me to have it run `run_keybase` on boot?

**Comment by tcby45** [comment]
Actually, I lied (sorry). `/keybase/public` and `/keybase/private` exist, along with a `/keybase/kbfs.nologin.txt`, but attempting to cd anywhere in `public` results in the terminal hanging and having to be manually closed out. `run_keybase` displays the usual, but `keybase login` literally does nothing and goes back to the prompt: I would tell you what exit code occurs, but `$?` doesn't work for some reason.

**Comment by cjb** [comment]
It's probably trying to use the GUI, which isn't working because of the `libnotify` error.

Is this a desktop machine or a headless server?

**Comment by tcby45** [comment]
Headless - like I said, worked fine before.

**Comment by cjb** [comment]
Try `keybase unlock`

**Comment by tcby45** [comment]
That was successful, but attempting to `cd /keybase/public/readyp1` still hung the terminal, resulting in me manually closing the terminal out. At this rate, should I just uninstall it and reinstall?

**Comment by strib** [comment]
Probably no need to reinstall, likely something fixable.  Can you try `keybase log send` again and report the log ID?

**Comment by tcby45** [comment]
It's df28f30b95610a255f5ab51c

Also, I just noticed when I ran it that it gave me a couple of errors at the top:

```sh
▶ WARNING error opening log "/home/ubuntu/.cache/keybase/keybase.updater.log": open /home/ubuntu/.cache/keybase/keybase.updater.log: no such file or directory
▶ WARNING error opening log "/home/ubuntu/.cache/keybase/keybase.start.log": open /home/ubuntu/.cache/keybase/keybase.start.log: no such file or directory
```

**Comment by strib** [comment]
Hi @tcby45, is it working now?  It looks like to me that you successfully unlocked your key when you ran `keybase unlock`, but then KBFS was in a exponential backoff where it was waiting about 90 seconds in between connection attempts.  You tried to access `/keybase/public/readyp1` within that 90 seconds, and it hung.  But after that, it looks like KBFS successfully connected to the backend server, so I'd expect future accesses to work.

Clearly this is a bad user experience, and we should try to short-circuit that backoff algorithm somehow once the key is unlocked.

Also, for what it's worth, it looks like you shell is very aggressive about looking stuff up in `$PWD`.  Basically every letter of every command you type ends up being a KBFS lookup, which is probably going to be very slow for you.  You might want to disable those parts of your shell, including having `.` in your `$PATH`, while you're roaming around KBFS.
